src/pages/dashboard/settings.tsx

```diff
@@ -1,9 +1,189 @@
   const { user } = useUser({ redirectTo: "/login" });
-  return <></>;
+  const { data: donationSettings } = useDonationSettings(user?.name);
+  const { data: walletSettings } = useWalletSettings(user?.id);
+
+  const queryClient = useQueryClient();
+
+  const addDonationSettings = useMutation(
+    (donationSettings) => {
+      const body = { donationSettings };
+      return fetchJson(`/api/donation-settings/update/${user?.id}`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(body),
+      });
+    },
+    {
+      onMutate: async (settings: Donation_settings) => {
+        await queryClient.cancelQueries(["streamer", user?.name]);
+        const previousSettings = queryClient.getQueryData<Donation_settings>([
+          "streamer",
+          user?.name,
+        ]);
+
+        console.log({ settings });
+
+        if (previousSettings) {
+          queryClient.setQueryData<Donation_settings>(
+            ["streamer", user?.name],
+            { ...previousSettings }
+          );
+        }
+
+        return { previousSettings };
+      },
+      onError: (err, variables, context) => {
+        if (context?.previousSettings) {
+          queryClient.setQueryData<Donation_settings>(
+            ["streamer", user?.name],
+            context.previousSettings
+          );
+        }
+      },
+      onSettled: () => {
+        queryClient.invalidateQueries(["streamer", user?.name]);
+      },
+    }
+  );
+  const addWalletSettings = useMutation(
+    (walletSettings) => {
+      const body = { walletSettings };
+      return fetchJson(`/api/wallet/settings/update/${user?.id}`, {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(body),
+      });
+    },
+    {
+      onMutate: async (settings: Wallet) => {
+        await queryClient.cancelQueries(["streamer", user?.id]);
+        const previousSettings = queryClient.getQueryData<Wallet>([
+          "streamer",
+          user?.id,
+        ]);
+
+        console.log({ settings });
+
+        if (previousSettings) {
+          queryClient.setQueryData<Wallet>(["streamer", user?.id], {
+            ...previousSettings,
+          });
+        }
+
+        return { previousSettings };
+      },
+      onError: (err, variables, context) => {
+        if (context?.previousSettings) {
+          queryClient.setQueryData<Wallet>(
+            ["streamer", user?.id],
+            context.previousSettings
+          );
+        }
+      },
+      onSettled: () => {
+        queryClient.invalidateQueries(["streamer", user?.id]);
+      },
+    }
+  );
+
+  const handleWalletSettingsSubmit = async (
+    event: FormEvent<HTMLFormElement>
+  ) => {
+    event.preventDefault();
+    if (!user) return;
+
+    const data = new FormData(event.currentTarget);
+    // TODO do we need to validate?
+
+    let updateData = {};
+
+    for (const pair of data.entries()) {
```

Comment:
This looks very complicated. Can you explain the implementation of this for-loop? Is there any way to simplify this?